src/pylibsshext/session.pyx

```diff
@@ -46,11 +46,11 @@ OPTS_DIR_MAP = {
 
 LOG_MAP = {
     logging.NOTSET: libssh.SSH_LOG_NONE,
-    logging.DEBUG: libssh.SSH_LOG_DEBUG,
```

Comment:
This doesn't sound right. Why not just add another level called `TRACE` mapped to a low numeric value like `5`? (`DEBUG` is `10` in stdlib)

Reply:
I am not much knowledgeable about python so sorry if I miss something.

How would callers discover and set such custom log level?

As I can see from `ansible-collections/ansible.netcommon` this is called like
```
            self.ssh.set_log_level(logging.INFO)
```
Would it mean we should define a new list of log levels that could be used instead of the standard one? I can certainly do that if it would be preferred. I would consider it as quite a barrier to cross for the users updating from older version to learn there is one more non-standard log level and how to set it.

This was an attempt to provide full mapping of standard log levels to libssh log levels  for the price of lower granularity on the verbose end (the difference between DEBUG and TRACE is not that significant, but the later is much more useful).

If you want full granularity, maybe placing the libssh `SSH_LOG_DEBUG` between the `DEBUG` and `INFO` with value lets say `15` might be better to simplify the barrier to set up the most verbose debug level?

Reply:
I'm leaning towards custom log level, but a precise one.

There's a warning recommending not to add custom log levels to libraries, though: https://docs.python.org/3/howto/logging.html#custom-levels.

Somewhere on the page, it's also suggested that https://docs.python.org/3/library/logging.html#logging.Logger.log should be used for such cases.

And https://docs.python.org/3/library/logging.html#logging.addLevelName exists for declaring additional log levels.

I wonder if we should declare some `ANSIBLE_PYLIBSSH_`-prefixed levels for custom in-between stuff. We could then also expose importable constants that the end-users of the library would be able to use w/o clushing with stdlib. This would address the concern of that stdlib docs warning.

I don't think we should shift what `DEBUG` means. And `SSH_LOG_DEBUG` shouldn't be reassigned to "something between debug and info". Instead, a new `ANSIBLE_PYLIBSSH_TRACE` could exist that would be represented by a number below `10`.

Reply:
I don't know. I am not that much of a python programmer. The first linked paragraph explicitly discourages the use of custom log levels in libraries, but I did not dive into that deep enough.

My reasoning for shifting the stdlib DEBUG to libssh TRACE is that the DEBUG is the most verbose logging in stdlib and therefore it should map to the most verbose log level in libssh. If it does not, setting the "most verbose" log level becomes more complicated for users/developers using the pylibssh and debugging further issues that require more verbose log will become harder for us, libssh developers.

So if you have some idea/strong opinion how this should work, please, feel free to propose the changes.